core/dbt/parser/schemas.py

```diff
@@ -100,8 +100,11 @@
 def yaml_from_file(source_file: SchemaSourceFile) -> Dict[str, Any]:
     """If loading the yaml fails, raise an exception."""
     path = source_file.path.relative_path
+    # File contents filed is Optional, and is sometimes None, so ensure
+    # we are not passing in None to load_yaml_text.
+    contents = source_file.contents if source_file.contents else ""
```

Comment:
I can go either way on this, but this could be:
```contents = source_file.contents or ""```

Taken further, this function could be:
```
def yaml_from_file(source_file: SchemaSourceFile) -> Dict[str, Any]:
    try:
        return load_yaml_text(source_file.contents or "", source_file.path)
    except ValidationException as e:
        raise YamlLoadFailure(source_file.project_name, source_file.path.relative_path, e)
```
The initial variables of `path` and `contents` are only used once, so the only benefit is readability. However, in both cases I need to refer to the variable definition to see what's being passed in; i.e. `contents` is forced to have a value, hence is not quite the same as source_file.contents, and `path` is actually the relative path, and not the absolute path. If these are moved into the function call, it's more readable to me (though that may differ by user). Also, the docstring is basically the code at that point, so it's no longer needed.

As an alternative for the None vs "" issue with `load_yaml_text()`, should we consider overriding None with "" in `load_yaml_text()` itself? In other words, will this happen elsewhere? Will we always want to pass in an empty string when we have a None? Or are there valid scenarios where we want to pass in a None for this argument and generate the resulting error?

Reply:
I changed yaml_from_file as per your suggestion. I'd rather not change load_yaml_text at this point because we're going to backport this and we want to make backport changes as minimal as possible.